### ISSUE TITLE
ViewComponents with non-primitive parameters.

### ISSUE THREAD

**Issue by theonlylawislove** [aspnet/Mvc#3720]
```c#
public IViewComponentResult Invoke(string subName, MyNamespace.MyInstance instance)
```

When attempting to render this view component, no matter what, I get this:

> Could not find an 'Invoke' method matching the parameters.

If I change ```MyNamespace.MyInstance``` to ```System.String```, everything works.

Is this intended?

**Comment by theonlylawislove** [comment]
Also, if I pass in a nullable type (```System.String``` for example), I get this same error. If I set it to an empty string, it works.

What if someone wants ```int?``` or ```Guid?```?

**Comment by Muchiachio** [comment]
This may be a duplicate of https://github.com/aspnet/Mvc/issues/3211?

**Comment by theonlylawislove** [comment]
I'm using ```rc1-final```, and it still doesn't work. Also, I'd like support for custom classes/instances, not just nullable primitives.

**Comment by Eilon** [comment]
This is a dup of https://github.com/aspnet/Mvc/issues/612. In #3211 we fixed just one error case, but in #612 we're going to do a much bigger fix that should solve all the ugliness around invoking View Components.
